spark-submission-worker/src/main/java/org/apache/spark/k8s/operator/SparkClusterResourceSpec.java

```diff
@@ -45,20 +48,38 @@ public SparkClusterResourceSpec(SparkCluster cluster, SparkConf conf) {
     String namespace = conf.get(Config.KUBERNETES_NAMESPACE().key(), clusterNamespace);
     String image = conf.get(Config.CONTAINER_IMAGE().key(), "spark:4.0.0-preview1");
     ClusterSpec spec = cluster.getSpec();
-    masterService = buildMasterService(clusterName, namespace);
-    masterStatefulSet = buildMasterStatefulSet(clusterName, namespace, image);
+    masterService =
+        buildMasterService(
+            clusterName,
+            namespace,
+            cluster.getSpec().getMasterSpec().getMasterServiceMetadata(),
```

Comment:
nit. Just a coding style comment. It's not good to repeat the same thing multiple time. For example, we already have `spec` as a local variable. So, let's not add a new like this, `cluster.getSpec().getMasterSpec()`.